yarn-project/end-to-end/src/e2e_prover/e2e_prover_test_avm.ts

```diff
@@ -0,0 +1,242 @@
+import { SchnorrAccountContractArtifact, getSchnorrAccount } from '@aztec/accounts/schnorr';
+import {
+  type AccountWalletWithSecretKey,
+  type AztecNode,
+  type CompleteAddress,
+  type DebugLogger,
+  type Fq,
+  type Fr,
+  type PXE,
+  createDebugLogger,
+  deployL1Contract,
+} from '@aztec/aztec.js';
+import { BBCircuitVerifier } from '@aztec/bb-prover';
+import { RollupAbi } from '@aztec/l1-artifacts';
+import { AvmTestContract } from '@aztec/noir-contracts.js';
+import { type PXEService } from '@aztec/pxe';
+
+// @ts-expect-error solc-js doesn't publish its types https://github.com/ethereum/solc-js/issues/689
+import solc from 'solc';
+import { getContract } from 'viem';
+
+import { waitRegisteredAccountSynced } from '../benchmarks/utils.js';
+import { getACVMConfig } from '../fixtures/get_acvm_config.js';
+import { getBBConfig } from '../fixtures/get_bb_config.js';
+import {
+  type ISnapshotManager,
+  type SubsystemsContext,
+  addAccounts,
+  createSnapshotManager,
+  publicDeployAccounts,
+} from '../fixtures/snapshot_manager.js';
+import { setupPXEService } from '../fixtures/utils.js';
+
+const { E2E_DATA_PATH: dataPath } = process.env;
+
+const SALT = 1;
+
+type ProvenSetup = {
+  pxe: PXE;
+  teardown: () => Promise<void>;
+};
+
+/**
+ * Simpler version of FullProverTest.
+ */
+export class FullProverTestAvm {
```

Comment:
Can't pull some of this into a shared parent class with `FullProverTest`? Fine if not!

Reply:
Will this separate test class still exist after the AVM can prove everything?

Reply:
Ah I see, the difference isn't just that it's testing the AVM, it's that it's using the AVM test contract and just doing addArgsReturn. You can probably ignore these comments! I guess once the AVM can prove everything, we can use this class for the simple test, and the other (more full) class for the token test (possibly with modifications if necessary?)